File: ai/mcp/typescript/clients/MicrosoftLearn.ts
import { McpClient } from '../McpClient.js';
import {
    McpTool,
    McpPrompt,
    McpResource
} from '../McpTypes.js'

/**
 * Microsoft learn.
 * // https://github.com/microsoftdocs/mcp
 */
export class MicrosoftLearn extends McpClient {

    /**
     * Microsoft learn.
     */
    constructor() {
        super("MicrosoftDocsSearch", "1.8.9");
    }

    /**
     * open microsoft learn connection.
     * @returns {boolean} true if opened; else false.
     */
    async openMicrosoftLearn(): Promise<void> {

        try {
            // open a new connection.
            await this.openConnectionHttp("https://learn.microsoft.com/api/mcp");

            // change the required parameter
            let tools: McpTool[] = this.getTools();
            for (var i = 0; i < tools.length; i++) {
                if (!tools[i].parameters.required) {
                    // change
                    tools[i].parameters.required = ['query'];
                }
            }

        } catch (e) {
            let error: any = e;
        }
    }
}


//var client = new MicrosoftLearn();
//await client.openMicrosoftLearn();
//var tools = client.getTools();
//console.log(tools[2].parameters);

File: ai/mcp/python/clients/MicrosoftLearn.py
from typing import Optional, Any, List, Union

from ..McpClient import McpClient
from ..McpTypes import McpTool

# Microsoft learn.
class MicrosoftLearn(McpClient):
    """
    Microsoft learn.
    """
    def __init__(self):
        super().__init__()

    async def openMicrosoftLearn(self) -> None:
        """
        open microsoft learn connection.
        """
        try:
            # open a new connection.
            await self.openConnectionHttp("https://learn.microsoft.com/api/mcp")

            # change the required parameter
            tools: List[McpTool] = self.getTools()
            for tool in tools:
                if tool.parameters.parameters["required"] is None:
                    # change
                    tool.parameters.parameters["required"] = ['query', 'question']

        except Exception as e:
            if (self.logEvent):
                self.logEvent("error", "openmicrosoftlearn", "could not open the connection", e)